src/mds/Server.cc

```diff
@@ -2531,7 +2531,36 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
     respond_to_request(mdr, mdr->more()->peer_error);
     return;
   }
-  
+
+  Session *session = mds->get_session(req);
+  if (!session->info.has_feature(CEPHFS_FEATURE_ALTERNATE_NAME)) {
+    if (req->get_op() == CEPH_MDS_OP_CREATE ||
+	req->get_op() == CEPH_MDS_OP_SETXATTR ||
+	req->get_op() == CEPH_MDS_OP_SETATTR ||
+	req->get_op() == CEPH_MDS_OP_MKNOD ||
+	req->get_op() == CEPH_MDS_OP_LINK ||
+	req->get_op() == CEPH_MDS_OP_RENAME ||
+	req->get_op() == CEPH_MDS_OP_MKDIR ||
+	req->get_op() == CEPH_MDS_OP_SYMLINK ||
+	req->get_op() == CEPH_MDS_OP_MKSNAP) {
```

Comment:
Occasionally we add new ops, so this sort of blocklist may be hard to keep up with. Could we instead just specifically allow certain ops for these clients? In particular: GETATTR, READDIR, UNLINK, and RMDIR.

Anything else, I think we'll want to disallow.

Reply:
...and maybe GETVXATTR as well.